atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/ValueType.java

```diff
@@ -840,6 +842,77 @@ public String getPersistCode(String variableName) {
             return Long.class;
         }
     },
+    UUID {
+        @Override
+        public UUID convertToJava(byte[] value, int offset) {
+            Preconditions.checkArgument(value.length == 16);
```

Comment:
This isn't necessarily true, the value could have stuff other than the uuid packed inside it.